run.py

```diff
@@ -3,27 +3,31 @@
     import os
     import sys
 
-    if sys.version_info[0:2] < (3, 4):
-        raise SystemExit('python 3.4+ is required')
+    if sys.version_info[0:2] < (3, 6):
+        raise SystemExit('Python 3.6+ is required')
+
+    root_path = os.path.abspath(os.path.dirname(__file__))
 
     try:
         import mtp_common
 
-        if mtp_common.VERSION < (5,):
+        # NB: this version does not need to be updated unless mtp_common changes significantly
+        if mtp_common.VERSION < (10,):
             raise ImportError
     except ImportError:
         try:
-            try:
-                from pip._internal import main as pip_main
-            except ImportError:
-                from pip import main as pip_main
+            import pkg_resources
         except ImportError:
             raise SystemExit('setuptools and pip are required')
+        try:
+            pip = pkg_resources.load_entry_point('pip', 'console_scripts', 'pip')
+        except pkg_resources.ResolutionError:
+            raise SystemExit('setuptools and pip are required')
 
-        print('Pre-installing MTP-common')
-        pip_main(['--quiet', 'install', '--upgrade', 'money-to-prisoners-common'])
+        print('Pre-installing MTP-common and base requirements')
+        pip(['install', '--requirement', f'{root_path}/requirements/base.txt'])
```

Comment:
Do we know that ` -r <file>` is equivalent to `--upgrade --upgrade-strategy=only-if-needed -r <file>` for both the upgrade and downgrade case?

Reply:
i believe upgrades and downgrades are treated the same by `pip`, simply as a version mismatch

Reply:
but this pip invocation doesn't try to upgrade installed dependencies that satisfy the requirements files. i.e. it's
```
pip install --upgrade-strategy=only-if-needed -r <file>
```
not
```
pip install --upgrade --upgrade-strategy=only-if-needed -r <file>
```